### PR TITLE
ONe network test refactoring: request

### PR DESCRIPTION
**Pull request by pauloewerton**
this change adds the ONe network plugin requestInstance method related tests refactoring. to test: check out code; run requestIntance unit tests. notice this should not be reviewed until pr #339 is merged.

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/network/v5_4/OpenNebulaNetworkPlugin.java [hunk 1]
```diff
@@ -79,50 +81,26 @@ public boolean hasFailed(String cloudState) {
...
-		if (addressRangeIndex != null) {
-			addressRangeId = virtualNetwork.xpath(String.format(ADDRESS_RANGE_ID_PATH_FORMAT, addressRangeIndex));
-		} else {
-			throw new NoAvailableResourcesException(String.format(Messages.Exception.UNABLE_TO_CREATE_NETWORK_RESERVE,
-					networkOrder.getCidr()));
-		}
 
+		Integer addressRangeIndex = this.getAddressRangeIndex(virtualNetwork, subnetInfo.getLowAddress(), size);
+		String addressRangeId = this.getAddressRangeId(virtualNetwork, addressRangeIndex, cidr);
 		String ip = this.getNextAvailableAddress(virtualNetwork, addressRangeIndex);
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/network/v5_4/OpenNebulaNetworkPlugin.java]: My suggestion is that these lines (from 84 to 92) be migrated to the _getCreateNetworkReserveRequest_ method which would only contain the _virtualNetwork_ parameter.

Comment (pauloewerton) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/network/v5_4/OpenNebulaNetworkPlugin.java]: done

**Comment by pauloewerton** [comment]
@wassoares thank you for you review. comment addressed.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/network/v5_4/OpenNebulaNetworkPluginTest.java [hunk 2]
```diff
@@ -45,71 +43,246 @@
...
+	    // set up
+		String expectedMessage = String.format(Messages.Exception.UNABLE_TO_CREATE_NETWORK_RESERVE, FAKE_CIDR_ADDRESS);
+
+		// exercise
+		try {
+			this.plugin.getAddressRangeId(this.virtualNetwork, null, FAKE_CIDR_ADDRESS);
+			Assert.fail();
+		} catch (NoAvailableResourcesException e) {
+			// verify
+            Assert.assertEquals(expectedMessage, e.getMessage());
```
[the remark this answers is not in the document]
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/network/v5_4/OpenNebulaNetworkPluginTest.java]: fixed

**Pull request merged**